new-examples/graphql-service-oauth2/graphql_service_oauth2.bal

```diff
@@ -0,0 +1,44 @@
+import ballerina/graphql;
+
+listener graphql:Listener securedEP = new(9090,
+    secureSocket = {
+        key: {
+            certFile: "../resource/path/to/public.crt",
+            keyFile: "../resource/path/to/private.key"
+        }
+    }
+);
+
+// The service can be secured with OAuth2 authentication and can be authorized
+// optionally. OAuth2 authentication can be enabled by setting the
+// `graphql:OAuth2IntrospectionConfig` configurations.
+// Authorization is based on scopes. A scope maps to one or more groups.
+// Authorization can be enabled by setting the `string|string[]` type
+// configurations.
+@graphql:ServiceConfig {
+    auth: [
+        {
+            oauth2IntrospectionConfig: {
+                url: "https://localhost:9445/oauth2/introspect",
+                tokenTypeHint: "access_token",
+                scopeKey: "scp",
+                clientConfig: {
+                    customHeaders: {"Authorization": "Basic YWRtaW46YWRtaW4="},
+                    secureSocket: {
+                        cert: "../resource/path/to/public.crt"
+                    }
+                }
+            },
+            scopes: ["admin"]
+        }
+    ]
+}
+service /graphql on securedEP {
+
+    // It is optional to override the authentication and authorization
+    // configurations at the resource levels. Otherwise, the service auth
+    // configurations will be applied automatically to the resources as well.
```

Comment:
Is this supported in GraphQL?

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-distribution/pull/1816/commits/60e9f102c550acf74377bc2d648196efd76da06d